src/test/feefrac_tests.cpp

```diff
@@ -17,6 +17,27 @@ BOOST_AUTO_TEST_CASE(feefrac_operators)
     FeeFrac empty{0, 0};
     FeeFrac zero_fee{0, 1}; // zero-fee allowed
 
+    BOOST_CHECK_EQUAL(zero_fee.Evaluate(0), 0);
+    BOOST_CHECK_EQUAL(zero_fee.Evaluate(1), 0);
+    BOOST_CHECK_EQUAL(zero_fee.Evaluate(1000000), 0);
+    BOOST_CHECK_EQUAL(zero_fee.Evaluate(0x7fffffff), 0);
+
+    BOOST_CHECK_EQUAL(p1.Evaluate(0), 0);
+    BOOST_CHECK_EQUAL(p1.Evaluate(1), 10);
+    BOOST_CHECK_EQUAL(p1.Evaluate(100000000), 1000000000);
+    BOOST_CHECK_EQUAL(p1.Evaluate(0x7fffffff), int64_t(0x7fffffff) * 10);
+
+    FeeFrac neg{-1001, 100};
+    BOOST_CHECK_EQUAL(neg.Evaluate(0), 0);
+    BOOST_CHECK_EQUAL(neg.Evaluate(1), -11);
```

Comment:
how often would we get negative fee in reality? If rounding towards negative infinity just a theoretical exercise or is it an important usecase?